### DISCUSSION TITLE
Black Screen Streaming to Icecast

### DISCUSSION THREAD

**Discussion by ralphotowo** [openbroadcaster/obplayer#56] [Show and tell]
Hi, 

I have observer version 4 installed on Server A and obplayer version 5 installed on Server B. I wrote a script to install [obplayer5](https://www.dropbox.com/s/o3kcebumbgbj7rp/obplayer_setup.sh?dl=0) and the required dependencies because I'll eventually require lots of players on different VMs.

I'm able to stream audio without issue, however video streams only display a black screen while the audio plays. I've tried different types of video formats. The following is the debug log output from the player. Please let me know if I missed something, or if you have any suggestions.

> [Feb 28 2018 16:13:30 UTC] [module] loading module player
> [Feb 28 2018 16:13:30 UTC] [debug] adding element to bin: audio-out-capsfilter
> [Feb 28 2018 16:13:30 UTC] [debug] adding element to bin: audio-out-level
> [Feb 28 2018 16:13:30 UTC] [debug] adding element to bin: audio-out-interlink-tee
> [Feb 28 2018 16:13:30 UTC] [debug] adding element to bin: audio-out-post-tee-queue
> [Feb 28 2018 16:13:30 UTC] [debug] adding element to bin: audiosink
> [Feb 28 2018 16:13:30 UTC] [debug] adding element to bin: image-pipeline-freeze
> [Feb 28 2018 16:13:30 UTC] [debug] adding element to bin: image-pipeline-scale
> [Feb 28 2018 16:13:30 UTC] [debug] adding element to bin: image-pipeline-convert
> [Feb 28 2018 16:13:30 UTC] [debug] adding element to bin: image-pipeline-rate
> [Feb 28 2018 16:13:30 UTC] [debug] adding element to bin: image-pipeline-capsfilter
> [Feb 28 2018 16:13:30 UTC] [debug] adding element to bin: image-pipeline-alpha
> [Feb 28 2018 16:13:30 UTC] [debug] adding element to bin: image-pipeline-videomixer
> [Feb 28 2018 16:13:30 UTC] [debug] adding element to bin: rtp-input-pre-queue
> [Feb 28 2018 16:13:30 UTC] [debug] adding element to bin: rtp-input-depay
> [Feb 28 2018 16:13:30 UTC] [debug] adding element to bin: rtp-input-decode
> [Feb 28 2018 16:13:30 UTC] [debug] adding element to bin: rtp-input-convert
> [Feb 28 2018 16:13:30 UTC] [debug] adding element to bin: rtp-input-resample
> [Feb 28 2018 16:13:30 UTC] [debug] adding element to bin: rtp-input-post-queue
> [Feb 28 2018 16:13:30 UTC] [module] loading module httpadmin
> [Feb 28 2018 16:13:30 UTC] [debug] starting thread &lt;HTTPAdminThread&gt;
> [Feb 28 2018 16:13:30 UTC] [module] loading module testsignal
> [Feb 28 2018 16:13:30 UTC] [module] loading module fallback
> [Feb 28 2018 16:13:30 UTC] [admin] serving http(s) on port 23233
> [Feb 28 2018 16:13:30 UTC] [module] loading module scheduler
> [Feb 28 2018 16:13:30 UTC] [data] restoring database from file
> [Feb 28 2018 16:13:30 UTC] [data] done restoring database
> [Feb 28 2018 16:13:30 UTC] [data] resetting show, media, and priority data
> [Feb 28 2018 16:13:30 UTC] [debug] starting thread &lt;VersionUpdateThread&gt;
> [Feb 28 2018 16:13:30 UTC] [sync] sending player version to server: 5.0.0-20180119
> [Feb 28 2018 16:13:30 UTC] [sync] fetching show data from server
> [Feb 28 2018 16:13:31 UTC] [warning] server did not report a version number
> [Feb 28 2018 16:13:31 UTC] [sync] writing data to database
> [Feb 28 2018 16:13:32 UTC] [data] backup database to disk
> [Feb 28 2018 16:13:32 UTC] [data] done backing up
> [Feb 28 2018 16:13:32 UTC] [sync] fetching priority broadcast data from server
> [Feb 28 2018 16:13:32 UTC] [debug] starting thread &lt;SyncShowsThread&gt;
> [Feb 28 2018 16:13:32 UTC] [sync] fetching show data from server
> [Feb 28 2018 16:13:32 UTC] [debug] starting thread &lt;SyncEmergThread&gt;
> [Feb 28 2018 16:13:32 UTC] [debug] starting thread &lt;SyncMediaThread&gt;
> [Feb 28 2018 16:13:32 UTC] [debug] starting thread &lt;SyncPlaylogThread&gt;
> [Feb 28 2018 16:13:32 UTC] [module] loading module streamer
> [Feb 28 2018 16:13:32 UTC] [debug] starting thread &lt;PlayerThread&gt;
> [Feb 28 2018 16:13:32 UTC] [scheduler] loading show 0
> [Feb 28 2018 16:13:32 UTC] [scheduler] starting at track number 1
> [Feb 28 2018 16:13:32 UTC] [player] now playing track 2: Sample Video - MP4 Output (id: 22 file: 22-Sample_Video-MP4_Output.mov duration: 216.085s type: 'video' source: scheduler)
> [Feb 28 2018 16:13:32 UTC] [debug] video-playbin: patching audio/visual
> [Feb 28 2018 16:13:32 UTC] [debug] video-playbin: stopped by execute request
> [Feb 28 2018 16:13:32 UTC] [player] resuming track at 186.33302187919617 seconds.
> [Feb 28 2018 16:13:33 UTC] [debug] video-playbin: stopped
> [Feb 28 2018 16:13:33 UTC] [debug] video-playbin: stopped
> [Feb 28 2018 16:13:33 UTC] [debug] adding element to bin: queue2-0
> [Feb 28 2018 16:13:33 UTC] [debug] adding element to bin: interaudiosink0
> [Feb 28 2018 16:13:33 UTC] [debug] 10.0.0.1 "GET / HTTP/1.1" 200 -
> [Feb 28 2018 16:13:33 UTC] [debug] adding element to bin: interaudiosrc0
> [Feb 28 2018 16:13:33 UTC] [debug] adding element to bin: capsfilter2
> [Feb 28 2018 16:13:33 UTC] [debug] adding element to bin: queue2-1
> [Feb 28 2018 16:13:33 UTC] [debug] adding element to bin: audioconvert0
> [Feb 28 2018 16:13:33 UTC] [debug] adding element to bin: audioresample0
> [Feb 28 2018 16:13:33 UTC] [debug] adding element to bin: capsfilter3
> [Feb 28 2018 16:13:33 UTC] [debug] adding element to bin: vorbisenc0
> [Feb 28 2018 16:13:33 UTC] [debug] adding element to bin: queue2-2
> [Feb 28 2018 16:13:33 UTC] [debug] adding element to bin: intervideosrc0
> [Feb 28 2018 16:13:33 UTC] [debug] adding element to bin: queue2-3
> [Feb 28 2018 16:13:33 UTC] [debug] adding element to bin: videorate0
> [Feb 28 2018 16:13:33 UTC] [debug] adding element to bin: videoconvert0
> [Feb 28 2018 16:13:33 UTC] [debug] adding element to bin: videoscale0
> [Feb 28 2018 16:13:33 UTC] [debug] adding element to bin: videocapsfilter
> [Feb 28 2018 16:13:33 UTC] [debug] adding element to bin: theoraenc0
> [Feb 28 2018 16:13:33 UTC] [debug] adding element to bin: queue2-4
> [Feb 28 2018 16:13:33 UTC] [debug] adding element to bin: oggmux0
> [Feb 28 2018 16:13:33 UTC] [debug] adding element to bin: queue2-5
> [Feb 28 2018 16:13:33 UTC] [debug] adding element to bin: shout2send
> [Feb 28 2018 16:13:33 UTC] [debug] starting icecast streamer
> [Feb 28 2018 16:13:33 UTC] [debug] 10.0.0.1 "GET /common.css HTTP/1.1" 200 -
> [Feb 28 2018 16:13:33 UTC] [sync] writing data to database
> [Feb 28 2018 16:13:33 UTC] [debug] 10.0.0.1 "GET /extras/jquery.min.js HTTP/1.1" 200 -
> [Feb 28 2018 16:13:34 UTC] [data] backup database to disk
> [Feb 28 2018 16:13:34 UTC] [debug] 10.0.0.1 "GET /extras/leaflet.css HTTP/1.1" 200 -
> [Feb 28 2018 16:13:34 UTC] [debug] 10.0.0.1 "GET /common.js HTTP/1.1" 200 -
> [Feb 28 2018 16:13:34 UTC] [debug] 10.0.0.1 "GET /extras/leaflet.js HTTP/1.1" 200 -
> [Feb 28 2018 16:13:34 UTC] [debug] 10.0.0.1 "GET /extras/leaflet.draw.js HTTP/1.1" 200 -
> [Feb 28 2018 16:13:34 UTC] [data] done backing up
> [Feb 28 2018 16:13:34 UTC] [debug] 10.0.0.1 "GET /map.js HTTP/1.1" 200 -
> [Feb 28 2018 16:13:35 UTC] [debug] 10.0.0.1 "GET /extras/images/marker-shadow.png HTTP/1.1" 200 -
> [Feb 28 2018 16:13:35 UTC] [debug] 10.0.0.1 "GET /oblogo.svg HTTP/1.1" 200 -
> [Feb 28 2018 16:13:35 UTC] [debug] 10.0.0.1 "GET /extras/images/marker-icon.png HTTP/1.1" 200 -
> [Feb 28 2018 16:13:36 UTC] [debug] 10.0.0.1 "POST /strings HTTP/1.1" 200 -
> [Feb 28 2018 16:13:36 UTC] [debug] 10.0.0.1 "GET /favicon.ico HTTP/1.1" 200 -
> [Feb 28 2018 16:14:02 UTC] [debug] video-playbin: stopped by stop request
> [Feb 28 2018 16:14:03 UTC] [scheduler] loading show 0
> [Feb 28 2018 16:14:03 UTC] [scheduler] starting at track number 0
> [Feb 28 2018 16:14:03 UTC] [player] now playing track 1: Sample Video - Man Playing with Kids (id: 23 file: 23-Sample_Video-Man_Playing_with_Kids.mov duration: 16.267s type: 'video' source: scheduler)
> [Feb 28 2018 16:14:03 UTC] [debug] video-playbin: stopped by execute request
> [Feb 28 2018 16:14:03 UTC] [player] resuming track at 0.28513503074645996 seconds

**Comment by radiorob** [comment]
Showing something playing.  Have you tried the obvious and Enable Video Output? Adjusted any of the settings that come up? Does it give you the option to set aspect ratio?  Might have to reboot machine when changing video modes?  Are you running headless or is it installed on a desktop?

**Comment by ralphotowo** [comment]
> Are you running headless or is it installed on a desktop? 

It's installed on a virtual machine and it's running in headless mode. I'm starting obplayer with this command `obplayer_loop -d -r -H`.

> Might have to reboot machine when changing video modes?

Yes, I tried rebooting the server after updating the settings. 

> Have you tried the obvious and Enable Video Output?

I believe I have, see the following screenshot. 
![screenshot from 2018-03-02 09-45-17](https://user-images.githubusercontent.com/1202767/36891202-11f0e24e-1e01-11e8-80d1-1c518de07d19.png)

> Adjusted any of the settings that come up? Does it give you the option to set aspect ratio?

My focus has been on the streaming tab since I'm not looking to play the videos locally, I only want them streamed to icecast. This is working for audio with the current settings. 
![screenshot from 2018-03-02 09-45-53](https://user-images.githubusercontent.com/1202767/36891259-439aeace-1e01-11e8-8570-805f514b7c95.png)

**Comment by transistorfet** [comment]
If you run the player with the -H option, it will completely disable the video subsystem, ignoring the enable video option in the dashboard. Remove this flag from the command line.
